### PR TITLE
Remove empty metrics.go file and method call

### PR DESCRIPTION
**Pull request by johscheuer**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR removes a code part that is not used or has no functionality. The `Register()` function of the `nodeipam` module is called but does nothing. In the code the `util/metrics` module is called to actually register the metric(s) if the `nodeipam` module: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/nodeipam/node_ipam_controller.go#L106

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/nodeipam/node_ipam_controller.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ import (
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/record"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (draveness) [inline pkg/controller/nodeipam/node_ipam_controller.go]: remove `v1`

Comment (johscheuer) [reply pkg/controller/nodeipam/node_ipam_controller.go]: I changed this line back (seems to be a default vscode option)

**Comment by draveness** [comment]
Please squash the two commits into one. And ping approver after that.

**Comment by johscheuer** [comment]
Rebased like @draveness  mentioned

**Comment by johscheuer** [comment]
ping @bowei @gmarek

**Comment by dims** [comment]
/assign @smarterclayton @liggitt 

Clayton, Jordan, i don't see the current approvers active any more in the project, can you please `/approve`?

**Comment by jberkus** [comment]
Hey, even if this is unused code, it should have a (very brief) relase note.

**Comment by liggitt** [comment]
> Hey, even if this is unused code, it should have a (very brief) relase note.

No release note is needed for this, there are no user-facing changes.

**Pull request merged**